docs/source/running-jobs/priorities.rst

```diff
@@ -0,0 +1,182 @@
+
+1. When the job queue is longer than 4x the number of CPUs on the SkyPilot controller, any new high priority jobs will not be able to preempt low priority jobs until they are at the front of the queue.
+2. Priority settings only apply within a Kubernetes cluster.
+3. Preemption behavior depends on your cluster's configuration and may preempt other pods in the cluster.
+
+For more information, refer to the `Kubernetes documentation on Pod Priority and Preemption <https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/>`_.
+
+.. _priorities-example:
+
+Working Example
+---------------
+
+Below we show an example run with two priority classes: ``high-priority`` and ``low-priority``.
+
+.. _priorities-example-priority-classes:
+
+Step 1: Create Priority Classes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Create two `priority classes <https://kubernetes.io/docs/concepts/scheduling-eviction/pod-priority-preemption/#priorityclass>`_ in your Kubernetes cluster:
+
+.. code-block:: yaml
+
+  # priorities.yaml
+  apiVersion: scheduling.k8s.io/v1
+  kind: PriorityClass
+  metadata:
+    name: high-priority
+  value: 1000000
+  globalDefault: false
+  description: "High priority class for critical jobs"
+  ---
+  apiVersion: scheduling.k8s.io/v1
+  kind: PriorityClass
+  metadata:
+    name: low-priority
+  value: 10000
+  globalDefault: true
+  description: "Low priority class for background jobs"
+
+A higher value indicates higher priority. You can create as many priority classes as you want.
+
+Apply these priority classes to your cluster:
+
+.. code-block:: bash
+
+  kubectl apply -f priorities.yaml
+
+.. _priorities-example-sky-pilot-jobs:
+
+Step 2: Setting Priorities in SkyPilot Jobs
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To assign priorities to your SkyPilot tasks, use the ``experimental.config_overrides.kubernetes.pod_config`` field in your task YAML. 
+
+We use two simple counter jobs in this example:
+
+.. code-block:: yaml
+
+  # high-priority-task.yaml
+  resources:
+    cloud: kubernetes
+    cpus: 4
+
+  run: |
+    python -c '
+    import time
+    for i in range(1000):
+        print(f"High priority counter: {i}")
+        time.sleep(1)
+    '
+
+  experimental:
+    config_overrides:
+      kubernetes:
+        pod_config:
+          spec:
+            priorityClassName: high-priority
+
+.. code-block:: yaml
+
+  # low-priority-task.yaml
+  resources:
+    cloud: kubernetes
+    cpus: 4
+
+  run: |
+    python -c '
+    import time
+    for i in range(1000):
+        print(f"Low priority counter: {i}")
+        time.sleep(1)
+    '
+
+  experimental:
+    config_overrides:
+      kubernetes:
+        pod_config:
+          spec:
+            priorityClassName: low-priority
+
+.. tip::
+   For this example, be sure to set the ``resources.cpu`` field such that once one job is running, there are no CPUs left for the other job in the cluster.
```

Comment:
A bit confused, since snippets above already set it.

Reply:
Ah I meant to say the user needs to adjust the number of CPUs requested depending on the CPUs available in the cluster to create conditions for preemption. Reworded to:

```
  For this example, resource contention is necessary to see the preemption behavior. Be sure to set the ``resources.cpu`` field such that once one job is running, there are no CPUs left for the other job in the cluster. 
   
  You can inspect the total number of CPUs in the cluster using ``kubectl get nodes``.
```